src/NUnitTestAdapter/NUnitEngine/DiscoveryConverter.cs

```diff
@@ -0,0 +1,389 @@
+        {
+            var className = child.Attribute(NUnitXmlAttributeNames.Classname)?.Value;
+            var methodName = child.Attribute(NUnitXmlAttributeNames.Methodname)?.Value;
+            var seedAtr = child.Attribute(NUnitXmlAttributeNames.Seed)?.Value;
+            var seed = seedAtr != null ? long.Parse(seedAtr) : 0;
+            var btf = ExtractSuiteBasePropertiesClass(child);
+            var tc = new NUnitDiscoveryTestCase(btf, tf, className, methodName, seed);
+            return tc;
+        }
+
+
+        public static NUnitDiscoveryTestFixture ExtractTestFixture(INUnitDiscoveryCanHaveTestFixture parent, XElement node,
+            string className)
+        {
+            var b = ExtractSuiteBasePropertiesClass(node);
+            var ts = new NUnitDiscoveryTestFixture(b, className, parent);
+            return ts;
+        }
+
+        private static NUnitDiscoveryGenericFixture ExtractGenericTestFixture(
+            NUnitDiscoveryCanHaveTestFixture parent,
+            XElement node, string className)
+        {
+            var b = ExtractSuiteBasePropertiesClass(node);
+            var ts = new NUnitDiscoveryGenericFixture(b, parent);
+            return ts;
+        }
+        private static NUnitDiscoverySetUpFixture ExtractSetUpTestFixture(
+            NUnitDiscoveryCanHaveTestFixture parent,
+            XElement node, string className)
+        {
+            var b = ExtractSuiteBasePropertiesClass(node);
+            var ts = new NUnitDiscoverySetUpFixture(b, className, parent);
+            return ts;
+        }
+        private static NUnitDiscoveryParameterizedTestFixture ExtractParameterizedTestFixture(
+            NUnitDiscoveryCanHaveTestFixture parent, XElement node, string className)
+        {
+            var b = ExtractSuiteBasePropertiesClass(node);
+            var ts = new NUnitDiscoveryParameterizedTestFixture(b, parent);
+            return ts;
+        }
+
+        private NUnitDiscoveryTestAssembly ExtractTestAssembly(XElement node, NUnitDiscoveryTestRun parent)
+        {
+            string d_type = node.Attribute(NUnitXmlAttributeNames.Type).Value;
+            if (d_type != "Assembly")
+                throw new DiscoveryException("Node is not of type assembly: " + node);
+            var a_base = ExtractSuiteBasePropertiesClass(node);
+            var assembly = new NUnitDiscoveryTestAssembly(a_base, parent);
+            parent.AddTestAssembly(assembly);
+            return assembly;
+        }
+
+        private static BaseProperties ExtractSuiteBasePropertiesClass(XElement node)
+        {
+            string dId = node.Attribute(NUnitXmlAttributeNames.Id).Value;
+            string dName = node.Attribute(NUnitXmlAttributeNames.Name).Value;
+            string dFullname = node.Attribute(NUnitXmlAttributeNames.Fullname).Value;
+            var dRunstate = ExtractRunState(node);
+            const char apo = (char)0x22;
```

Comment:
I was going to suggest `'\u0022'`, and then I realized what `apo` meant!

```suggestion
            const char apo = '\'';
```